compile.go

```diff
@@ -299,7 +301,7 @@ func evalBuiltInPackages() error {
 			return errors.Wrapf(err, codes.Inherit, "package has invalid builtins %q", astPkg.Path)
 		}
 
-		itrp := interpreter.NewInterpreter()
+		itrp := interpreter.NewInterpreter(context.TODO(), dependencies.NewDefaultDependencies())
```

Comment:
This can safely be `context.Background()`.

How will `NewDefaultDependencies` work? Does every dependency have to have a default?

What about a pattern where a `ErrorDependencies` is used for this case? My thinking is that if any of the builtins try and access the deps during startup they will not be available and we should error.

During `evalBuiltInPackages` no deps can/should be used. So concretely `http.post` will have a dep on a `SecretLookup` interface, but it doesn't use that dep unless `http.post` is called during startup which it should not be. 

Then on subsequent calls to `interpreter.Eval` new (fully initialized deps) can be provided and calling `http.post` will use those deps.

Reply:
Note: made the change here to `context.Background()` but still need to have discussion around the rest of this comment.